### PR TITLE
bpo-29673: Some gdb macros are broken in 3.6

### PR DESCRIPTION
**Pull request by ghost**
This PR fixes broken scripts for printing of Python call stack from GDB.

https://bugs.python.org/issue29673

### CODE REVIEW

File: Misc/NEWS.d/next/Tools-Demos/2018-03-16-17-25-05.bpo-29673.m8QtaW.rst [hunk 1]
```diff
@@ -0,0 +1,20 @@
+Fix gdbinit
```
Comment (ned-deily) [inline Misc/NEWS.d/next/Tools-Demos/2018-03-16-17-25-05.bpo-29673.m8QtaW.rst]: Thanks for including a NEWS entry.  However, this is a bit too detailed.  See a [current changelog](https://docs.python.org/3.7/whatsnew/changelog.html#changelog) to get a sense of the level of detail we use.  This kind of detail might be appropriate for a commit message.

**Review by smontanaro — COMMENTED** [review]
LGTM, modulo Ned's NEWS file comment. (I am really surprised this little hack has lived this long...)

File: Misc/ACKS [hunk 2]
```diff
@@ -1806,3 +1806,4 @@ Jelle Zijlstra
 Gennadiy Zlobin
 Doug Zongker
 Peter Åstrand
+Marcel Plch
```
Comment (ned-deily) [inline Misc/ACKS]: A nit: we try to keep the ACKs list in alphabetical order by family name.  So please insert your name in the "P" section.  Thanks!

File: Misc/NEWS.d/next/Tools-Demos/2018-03-16-17-25-05.bpo-29673.m8QtaW.rst [hunk 3]
```diff
@@ -0,0 +1,6 @@
+PyUnicode_AsString() was removed in Python and PyUnicode_AsUTF8() is now used.
```
Comment (ned-deily) [inline Misc/NEWS.d/next/Tools-Demos/2018-03-16-17-25-05.bpo-29673.m8QtaW.rst]: Thanks for condensing the NEWS entry.  But I think it is still too detailed and does not really answer the question of *what* was fixed; *why* it was fixed is not so important here.  Also, if you wish, you can add your name here.  An example from the [Developer's Guide](https://devguide.python.org/committing/#what-s-new-and-news-entries):

> Fix warning message when ``os.chdir()`` fails inside
> ``test.support.temp_cwd()``.  Patch by Chris Jerdonek.

**Review by ned-deily — CHANGES_REQUESTED** [review]
Thanks for the review, @smontanaro , and the changes, @Traceur759.  Almost there!

**Comment by Traceur759** [comment]
I have made the requested changes; please review again

**Review by ned-deily — APPROVED** [review]

**Pull request merged**